themes/aura/assets/upsell.js

```diff
@@ -24,24 +13,45 @@ document.getElementById("upsell-form").addEventListener("submit", async function
   };
 
   formData.forEach((value, key) => {
-    if (key.startsWith("product_offers")) {
-      const productId = key;
-      upsellParams.product_offers[productId] = value;
+    if (!["upsell_id", "order_id"].includes(key)) {
+      upsellParams.product_offers[key] = value;
     }
   });
 
+  await submitAnswer(event, upsellParams);
+});
+
+function loadingButton(event, loading) {
```

Comment:
We usually name functions with a verb instead of a noun, something like `setButtonIsLoading` is way more descriptive of the functions intent
```suggestion
function setButtonIsLoading(event, loading) {
```

Reply:
Since the function doesn't need the `event` and only needs the button (i.e event.target) we can send it when we call the function. That way the function has one responsibility